src/performance/gpu/asus/asus_wmi.rs

```diff
@@ -47,9 +73,31 @@ impl AsusWmi {
         let boost = self.boost().await?;
 
         // Set the STAPM value to the given TDP value
-        self.platform.set_ppt_pl1_spl(value as u8)?;
+        let val = AttrValue::Integer(value as i32);
+        if let Some(firmware_attributes) = &self.attributes {
+            match firmware_attributes
+                .attributes()
+                .iter()
+                .find(|a| a.name() == "ppt_pl1_spl")
+            {
+                Some(attribute) => match attribute.set_current_value(val) {
+                    Ok(_) => {
+                        log::info!("Set SPL to {value}");
+                    }
+                    Err(e) => {
+                        return Err(TDPError::FailedOperation(format!(
+                            "Failed to set SPL: {e:}"
+                        )));
+                    }
+                },
+
+                None => return Err(TDPError::FailedOperation("Failed to set SPL.".to_string())),
+            }
+        } else {
+            self.platform.set_ppt_pl1_spl(value as u8)?;
```

Comment:
I don't see how this case will ever be hit. If the "asus-wmi" module is found, the `attributes` property will always be `Some`.

Reply:
asus-wmi doesn't provide attributes, asus-armoury does. It depends on asus-wmi.

Reply:
Look at how the `AsusWmi` object is created in the constructor. If `RogPlatform::new()` successfully returns, then `self.attributes` is set to `Some`. So this case will never get hit.